### PR TITLE
[1.0] Ensure component file paths are normalized fixes #1093

### LINKED ISSUE #1093: [1.0] Gatsbygram Error: UNHANDLED REJECTION Invariant Violation: RelayCompilerContext: Duplicate document named `allImages`.
Gatsbygram is erroring out in the CI and not exiting. The full log is here: https://ci.appveyor.com/project/KyleAMathews/gatsby/build/1.0.370/job/40qel5sx059fbmyi

```
npm run build
> gatsby-example-image-gallery@1.0.0 build C:\projects\gatsby\examples\gatsbygram
> gatsby build
bin/gatsby: time since started: 0.097
bin/cli: time since started: 0.189
lib/bootstrap/index.js time since started: 1.203 sec
open and validate gatsby-config.js: 6.453ms
initial loading: 1590.370ms
copy gatsby files: 29.342ms
initial sourcing and transforming nodes: 534.850ms
(node:2552) Warning: No such label 'building schema' for console.timeEnd()
createPages: 149.637ms
createPagesStatefully: 14.281ms
UNHANDLED REJECTION Invariant Violation: RelayCompilerContext: Duplicate document named `allImages`. GraphQL fragments and roots must have unique names.
    at invariant (C:\projects\gatsby\examples\gatsbygram\node_modules\fbjs\lib\invariant.js:44:15)
    at RelayCompilerContext.add (C:\projects\gatsby\examples\gatsbygram\node_modules\relay-compiler\lib\RelayCompilerContext.js:61:34)
    at C:\projects\gatsby\examples\gatsbygram\node_modules\relay-compiler\lib\RelayCompilerContext.js:70:18
    at Array.reduce (native)
    at RelayCompilerContext.addAll (C:\projects\gatsby\examples\gatsbygram\node_modules\relay-compiler\lib\RelayCompilerContext.js:69:18)
    at Runner._callee3$ (C:\projects\gatsby\examples\gatsbygram\node_modules\gatsby\dist\internal-plugins\query-runner\query-compiler.js:198:51)
    at tryCatch (C:\projects\gatsby\examples\gatsbygram\node_modules\regenerator-runtime\runtime.js:65:40)
    at Generator.invoke [as _invoke] (C:\projects\gatsby\examples\gatsbygram\node_modules\regenerator-runtime\runtime.js:303:22)
    at Generator.prototype.(anonymous function) [as next] (C:\projects\gatsby\examples\gatsbygram\node_modules\regenerator-runtime\runtime.js:117:21)
    at step (C:\projects\gatsby\examples\gatsbygram\node_modules\babel-runtime\helpers\asyncToGenerator.js:17:30)
    at C:\projects\gatsby\examples\gatsbygram\node_modules\babel-runtime\helpers\asyncToGenerator.js:35:14
    at new Promise (C:\projects\gatsby\examples\gatsbygram\node_modules\core-js\library\modules\es6.promise.js:191:7)
    at Runner.<anonymous> (C:\projects\gatsby\examples\gatsbygram\node_modules\babel-runtime\helpers\asyncToGenerator.js:14:12)
    at Runner.write (C:\projects\gatsby\examples\gatsbygram\node_modules\gatsby\dist\internal-plugins\query-runner\query-compiler.js:235:22)
    at Runner._callee$ (C:\projects\gatsby\examples\gatsbygram\node_modules\gatsby\dist\internal-plugins\query-runner\query-compiler.js:110:55)
    at tryCatch (C:\projects\gatsby\examples\gatsbygram\node_modules\regenerator-runtime\runtime.js:65:40)
UNHANDLED REJECTION Invariant Violation: RelayCompilerContext: Duplicate document named `allImages`. GraphQL fragments and roots must have unique names.
    at invariant (C:\projects\gatsby\examples\gatsbygram\node_modules\fbjs\lib\invariant.js:44:15)
    at RelayCompilerContext.add (C:\projects\gatsby\examples\gatsbygram\node_modules\relay-compiler\lib\RelayCompilerContext.js:61:34)
    at C:\projects\gatsby\examples\gatsbygram\node_modules\relay-compiler\lib\RelayCompilerContext.js:70:18
    at Array.reduce (native)
    at RelayCompilerContext.addAll (C:\projects\gatsby\examples\gatsbygram\node_modules\relay-compiler\lib\RelayCompilerContext.js:69:18)
    at Runner._callee3$ (C:\projects\gatsby\examples\gatsbygram\node_modules\gatsby\dist\internal-plugins\query-runner\query-compiler.js:198:51)
    at tryCatch (C:\projects\gatsby\examples\gatsbygram\node_modules\regenerator-runtime\runtime.js:65:40)
    at Generator.invoke [as _invoke] (C:\projects\gatsby\examples\gatsbygram\node_modules\regenerator-runtime\runtime.js:303:22)
    at Generator.prototype.(anonymous function) [as next] (C:\projects\gatsby\examples\gatsbygram\node_modules\regenerator-runtime\runtime.js:117:21)
    at step (C:\projects\gatsby\examples\gatsbygram\node_modules\babel-runtime\helpers\asyncToGenerator.js:17:30)
    at C:\projects\gatsby\examples\gatsbygram\node_modules\babel-runtime\helpers\asyncToGenerator.js:35:14
    at new Promise (C:\projects\gatsby\examples\gatsbygram\node_modules\core-js\library\modules\es6.promise.js:191:7)
    at Runner.<anonymous> (C:\projects\gatsby\examples\gatsbygram\node_modules\babel-runtime\helpers\asyncToGenerator.js:14:12)
    at Runner.write (C:\projects\gatsby\examples\gatsbygram\node_modules\gatsby\dist\internal-plugins\query-runner\query-compiler.js:235:22)
    at Runner._callee$ (C:\projects\gatsby\examples\gatsbygram\node_modules\gatsby\dist\internal-plugins\query-runner\query-compiler.js:110:55)
    at tryCatch (C:\projects\gatsby\examples\gatsbygram\node_modules\regenerator-runtime\runtime.js:65:40)
```

### PR DESCRIPTION
**Pull request by KyleAMathews**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -44,3 +44,4 @@ node_modules/
 # IDE specific
 .vscode/
 yarn.lock
+package-lock.json
```
Comment (Saturate) [inline .gitignore]: The lock file should be committed with npm 5, this ensures that we all get the same packages. Same thing with yarn.lock

Comment (KyleAMathews) [inline .gitignore]: https://github.com/gatsbyjs/gatsby/pull/1070

We do commit the root yarn.lock file.https://github.com/gatsbyjs/gatsby/blob/1.0/yarn.lock

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit 0494210e1451b43799871793e30c08d3d6532c3e

https://app.netlify.com/sites/using-contentful/deploys/59370c9b424ef234e4bceecb

**Comment by KyleAMathews** [comment]
Deploy preview failed.

Built with commit 0494210e1451b43799871793e30c08d3d6532c3e

https://app.netlify.com/sites/image-processing/deploys/59370c9a424ef234e4bceec2

**Comment by KyleAMathews** [comment]
I think we've more-or-less got it! Appveyor got all the way through the tests + building Gatsbygram up to the static html build...  Pushed another commit removing my debugging code so let's see if that one gets through.

**Comment by KyleAMathews** [comment]
But I'm guessing/hoping that the stalling at building HTML was just some sort of Appveyor weirdness. Perhaps it hits a RAM limit or something.

**Comment by KyleAMathews** [comment]
Ughh it froze again. I'm guessing it is a RAM problem. We need to run the different webpack jobs in child processes so when they're done they free their memory. Anyways, that can wait for another day. Merging this now. @Saturate pretty sure this is working now but would love to hear from you as well!

**Comment by KyleAMathews** [comment]
Will also do a quick alpha release with this and writeup the CHANGELOG in the morning.

**Pull request merged**